### DISCUSSION TITLE
The search strategy does not return the correct values

### DISCUSSION THREAD

**Discussion by AriaAhmadi** [google/or-tools#3299] [CP-SAT questions] [answered]
Hi,

I hope you are well.

I am minimizing a loss for which there are several optimum points. I need the solver to return the solutions wherein the variables have the highest value. For example, if x \in {1,280} and if x = 10 and x = 280 both similarly result in the minimum loss, I'd like the solver to return x = 280 as the solution.  

For this purpose I add the line below. Although, I'm getting very small values for x.

x = model.NewIntVar(1 , 280 , 'x_1') 
model.AddDecisionStrategy([ x ], cp_model.CHOOSE_HIGHEST_MAX, cp_model.SELECT_MAX_VALUE) 

Q1: Am I using the strategy correctly for the said purpose? If not, what is the alternative?

Q2: If I can guess where the solution is in the hyper space (but am not sure about it), how can I hint the solver to search in that area first? Does that speed up the search?

I look forward to hearing from you.
Kind regards,
Aria.

**Comment by lperron** [comment, marked as answer]
1) Strategies are not a great way of specifying the prefered solution. It should be part of the objective.
2) By default, they are executed after the Boolean sub-part of the model (including all bool implies var == value, and bool implies var >= value) has reached the fix-point.
3) To force them to be executed faithfully, you need to specify the FIXED_SEARCH parameter.
